### DISCUSSION TITLE
How do I fix this command so --context-separator adds my newlines to output file?

### DISCUSSION THREAD

**Discussion by JJenkx** [BurntSushi/ripgrep#2268] [Q&A] [answered]
This command is sending matches to the output file without newlines. Does anyone know how I can fix it?

`find -name '*' -type f -regextype posix-extended -iregex '(^\.\/x\w\w\.All_Amazon_Meta\.json\.xz$)' | parallel -j+0 "xz -kd --stdout {} | LC_ALL=C rg -iP --context-separator '\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n' 'chapin.*viton'" >/home/jjenkx/amazon.matches.txt`

Edit:

`LC_ALL=C` I set because I saw somewhere online that it makes grep faster. I don't know what it really does. Should I drop it?

Upon more testing, --context-separator only seem to work when combined with -A -B

This doesn't add newlines to outfile
`cat textfile | rg -iP --context-separator '\n\n' 'chapin.*viton' >outfile`

This does add newlines to outfile
`cat textfile | rg -iP -A 25 -B 25 --context-separator '\n\n' 'chapin.*viton' >outfile`

**Comment by BurntSushi** [comment]
Could you please minimize it down to just an `rg` command? Or at least, provide an example input that reproduces to the problem.

Also, why are you setting `LC_ALL` before calling ripgrep?

**Reply by JJenkx** [reply]
See edit to above

**Comment by BurntSushi** [comment, marked as answer]
Apologies, but I think your question is just totally unclear. I would suggest that you try to outline the problem you're trying to solve.

For example, it looks like you are confused about what `--context-separator` is. Namely, you say, "--context-separator only seem to work when combined with -A -B." But this is confusing to me, because _of course_ it only works with `-A` and `-B`. That's exactly what it is documented to do. Maybe you haven't read the docs here?

```
--context-separator SEPARATOR
    The string used to separate non-contiguous context lines in the output. This is only used when one
    of the context flags is used (-A, -B or -C). Escape sequences like \x7F or \t may be used. The
    default value is --.

    When the context separator is set to an empty string, then a line break is still inserted. To
    completely disable context separators, use the --no-context-separator flag.
```

Also:

> `LC_ALL=C` I set because I saw somewhere online that it makes grep faster. I don't know what it really does. Should I drop it?

You heard correctly, and indeed, it is sometimes true. But that's for `grep`. You are using ripgrep. ripgrep is not grep. So advice given for grep doesn't necessarily apply to ripgrep.

To be fully clear here: setting `LC_ALL=C` has absolutely no impact on ripgrep.

More generally, I'd suggest you not just add stuff like `LC_ALL=C` to a command unless it actually provides a noticeable speedup.

**Reply by JJenkx** [reply]
You got me there. I haven't read the documentation in a long while. Thank you for the help
